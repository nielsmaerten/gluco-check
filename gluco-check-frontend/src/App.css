@import url("https://fonts.googleapis.com/css2?family=Poppins&display=swap");
@import url("https://fonts.googleapis.com/icon?family=Material+Icons");
