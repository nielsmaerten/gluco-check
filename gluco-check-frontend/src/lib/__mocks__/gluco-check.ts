import { plainToClass } from "class-transformer";
import { NightscoutValidationEndpointResponseDto } from "../NightscoutValidationClient/NightscoutValidationClientDto";

export const MOCK_NSV_RESPONSE_VALID = {
  token: {
    isValid: true,
    parsed: "token123",
  },
  url: {
    isValid: true,
    parsed: "https://example.com",
    pointsToNightscout: true,
  },
  nightscout: {
    glucoseUnit: "mmol",
    minSupportedVersion: "13.0.0",
    version: "14.0.7",
  },
  discoveredMetrics: [
    "blood sugar",
    "insulin on board",
    "carbs on board",
    "sensor age",
    "cannula age",
    "pump battery",
  ],
};

export const MOCK_NSV_RESPONSE_NON_NS_URL = {
  url: {
    parsed: "https://example.com",
    isValid: true,
    pointsToNightscout: false,
  },
  token: {
    parsed: "",
    isValid: false,
  },
  nightscout: {
    version: "",
    glucoseUnit: "",
    minSupportedVersion: "13.0.0",
  },
  discoveredMetrics: [],
};

export const MOCK_NSV_RESPONSE_NS_NEEDS_UPGRADE = {
  token: {
    isValid: false,
    parsed: "token123",
  },
  url: {
    isValid: true,
    parsed: "https://example.com",
    pointsToNightscout: true,
  },
  nightscout: {
    glucoseUnit: "mmol",
    minSupportedVersion: "13.0.0",
    version: "12.9.9",
  },
  discoveredMetrics: [],
};

export const MOCK_NSV_RESPONSE_INVALID_TOKEN = {
  url: {
    parsed: "https://example.com",
    isValid: true,
    pointsToNightscout: true,
  },
  token: {
    isValid: false,
    parsed: "token1234",
  },
  nightscout: {
    minSupportedVersion: "13.0.0",
    glucoseUnit: "mmol",
    version: "14.0.7",
  },
  discoveredMetrics: ["blood sugar"],
};

export const mockNsvResponseDtoValid = plainToClass(
  NightscoutValidationEndpointResponseDto,
  MOCK_NSV_RESPONSE_VALID
);

export const mockNsvResponseDtoNonNsUrl = plainToClass(
  NightscoutValidationEndpointResponseDto,
  MOCK_NSV_RESPONSE_NON_NS_URL
);

export const mockNsvResponseDtoNsNeedsUpgrade = plainToClass(
  NightscoutValidationEndpointResponseDto,
  MOCK_NSV_RESPONSE_NS_NEEDS_UPGRADE
);

export const mockNsvResponseDtoInvalidToken = plainToClass(
  NightscoutValidationEndpointResponseDto,
  MOCK_NSV_RESPONSE_INVALID_TOKEN
);
