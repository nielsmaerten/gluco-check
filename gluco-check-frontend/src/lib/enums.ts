export enum BloodGlucoseUnit {
  mgdl = "mg/dl",
  mmol = "mmol/L",
}

export enum NightscoutBloodGlucoseUnit {
  mgdl = "mg/dl",
  mmol = "mmol",
}

// @TODO: once this repo is properly part of the workspace,
// pull this in from gluco-check-core instead
// These are mapped to translations directly by case name,
// additions or changes here need to be reflected there
export enum DiabetesMetric {
  Everything = "everything",
  BloodSugar = "blood sugar",
  InsulinOnBoard = "insulin on board",
  CarbsOnBoard = "carbs on board",
  SensorAge = "sensor age",
  CannulaAge = "cannula age",
  PumpBattery = "pump battery",
  PumpReservoir = "pump reservoir",
}

export enum AvailableLanguage {
  English = "en",
  Dutch = "nl",
}
