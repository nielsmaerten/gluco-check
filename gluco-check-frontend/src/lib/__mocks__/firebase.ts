import { BloodGlucoseUnit, DiabetesMetric } from "../enums";
import { GlucoCheckUserDocument } from "../types";

export const mockUser: Partial<firebase.User> = {
  email: "example@example.com",
  uid: "uid123",
};

export const mockUserDocument: GlucoCheckUserDocument = {
  heardDisclaimer: false,
  defaultMetrics: [DiabetesMetric.BloodSugar],
  glucoseUnit: BloodGlucoseUnit.mgdl,
  nightscout: {
    url: "https://example.com",
    token: "nstoken123",
  },
};
