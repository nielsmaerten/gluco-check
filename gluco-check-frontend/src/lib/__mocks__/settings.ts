import { BloodGlucoseUnit, DiabetesMetric } from "../enums";
import { SettingsFormData } from "../types";

export const mockFormData: SettingsFormData = {
  defaultMetrics: [DiabetesMetric.BloodSugar],
  nightscoutToken: "I am a token",
  nightscoutUrl: "https://example.com",
  glucoseUnit: BloodGlucoseUnit.mgdl,
};
