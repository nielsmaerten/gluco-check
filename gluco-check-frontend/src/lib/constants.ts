import { BloodGlucoseUnit } from "./enums";

// firestore
export const FIRESTORE_FIELD_PATH_GLUCOSE_UNITS = "glucoseUnit";
export const FIRESTORE_FIELD_PATH_NIGHTSCOUT = "nightscout";
export const FIRESTORE_FIELD_PATH_DEFAULT_METRICS = "defaultMetrics";
export const FIRESTORE_FIELD_PATH_NIGHTSCOUT_URL = `${FIRESTORE_FIELD_PATH_NIGHTSCOUT}.url`;
export const FIRESTORE_FIELD_PATH_NIGHTSCOUT_TOKEN = `${FIRESTORE_FIELD_PATH_NIGHTSCOUT}.token`;
export const FIRESTORE_FIELD_HEARD_DISCLAIMER = "heardDisclaimer";

// gluco check
export const NIGHTSCOUT_VALIDATION_ENDPOINT_URL =
  process.env.REACT_APP_NIGHTSCOUT_VALIDATION_ENDPOINT_URL || "";

// configuration
export const ALERT_AUTOHIDE_DURATION = 7000;
export const VALIDATION_DEBOUNCE_DURATION = 500;
export const DEFAULT_ONBOARDING_CAROUSEL_INTERVAL = 5000;
export const DEFAULT_GLUCOSE_UNITS = BloodGlucoseUnit.mmol;

// urls
export const TERMS_AND_CONDITIONS_URL = "/terms";
export const HEALTH_DISCLAIMER_URL = "/terms";
export const PRIVACY_URL = "/terms";
export const NIGHTSCOUT_PROJECT_URL = "http://nightscout.info/";
export const FAQS_URL = "/faq";
export const GLUCO_CHECK_GITHUB_URL =
  "https://github.com/nielsmaerten/gluco-check";

export const APP_DEBUG = process.env.REACT_APP_DEBUG === "true" || false;
