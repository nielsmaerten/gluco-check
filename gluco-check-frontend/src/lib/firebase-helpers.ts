import { BloodGlucoseUnit, DiabetesMetric } from "./enums";
import { GlucoCheckUserDocument } from "./types";

export const FIRESTORE_DEFAULT_SET_OPTIONS: firebase.firestore.SetOptions = {
  mergeFields: [
    "defaultMetrics",
    "glucoseUnit",
    "nightscout.token",
    "nightscout.url",
  ],
};

export const DEFAULT_USER_DOCUMENT: GlucoCheckUserDocument = {
  defaultMetrics: [DiabetesMetric.BloodSugar],
  glucoseUnit: BloodGlucoseUnit.mgdl,
  nightscout: {
    token: "",
    url: "",
  },
};

export const getDocumentPathForUser = (user: firebase.User): string => {
  return `users/${user.email}`;
};
