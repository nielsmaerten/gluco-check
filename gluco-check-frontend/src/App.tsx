import React from "react";
import { auth } from "./lib/firebase";
import { useAuthState } from "react-firebase-hooks/auth";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect,
} from "react-router-dom";
import { getDocumentPathForUser } from "./lib/firebase-helpers";
import { useTranslation } from "react-i18next";
import {
  AppBar,
  Container,
  IconButton,
  makeStyles,
  Paper,
  Toolbar,
  Typography,
} from "@material-ui/core";
import { GitHub, Help } from "@material-ui/icons";

import Landing from "./pages/Landing";
import LanguageSelector from "./components/LanguageSelector";
import EditSettings from "./pages/EditSettings";
import Welcome from "./pages/Welcome";
import "./App.css";
import { FAQS_URL, GLUCO_CHECK_GITHUB_URL } from "./lib/constants";
import { AvailableLanguage } from "./lib/enums";

export const FirebaseUserDocumentContext = React.createContext("");

const useStyles = makeStyles((theme) => ({
  root: {
    "@media (prefers-reduced-motion: reduce)": {
      "& *": {
        animationDuration: "0.001ms !important",
        animationIterationCount: "1 !important",
        transitionDuration: "0.001ms !important",
      },
    },
  },
  container: {
    height: "100%",
  },
  surface: {
    padding: theme.spacing(2),
    marginBottom: theme.spacing(2),
    marginTop: theme.spacing(2),
  },
  leftToolbar: {},
  rightToolbar: {
    marginLeft: "auto", // switch to logical
    marginRight: -12,
  },
  nav: {
    "&, & > li": {
      margin: 0,
      padding: 0,
      listStyle: "none",
    },
    "& > li": {
      display: "inline-block",
    },
  },
  toolbar: {},
  navTitle: {
    "& a": {
      color: "white",
      textDecoration: "none",
    },
  },
}));

export default function App() {
  const classes = useStyles();
  const [user, loading] = useAuthState(auth);
  const { t, i18n } = useTranslation();
  const docPath = user ? getDocumentPathForUser(user) : ""; // TODO: we don't want this empty path to be possible, AND we want to auth protect any other authed routes

  let Content: React.ReactElement | null = null;
  if (user) {
    Content = <Welcome />;
  } else {
    if (!loading) {
      Content = <Landing />;
    }
  }

  const navigation = (
    <AppBar position="sticky">
      <Toolbar variant="regular" className={classes.toolbar}>
        <section className={classes.leftToolbar}>
          <Typography variant="h6" component="h1" className={classes.navTitle}>
            <Link to={`/${i18n.language}`}>{t("title")}</Link>
          </Typography>
        </section>
        <section className={classes.rightToolbar}>
          <ul className={classes.nav}>
            <li>
              <LanguageSelector />
            </li>
            <li>
              <IconButton
                aria-label={t("navigation.faqs")}
                color="inherit"
                data-testid="navigation-home"
                href={FAQS_URL}
              >
                <Help />
              </IconButton>
            </li>
            <li>
              <IconButton
                aria-label={t("navigation.github")}
                color="inherit"
                data-testid="navigation-contribute"
                href={GLUCO_CHECK_GITHUB_URL}
                target="_contribute"
              >
                <GitHub />
              </IconButton>
            </li>
          </ul>
        </section>
      </Toolbar>
    </AppBar>
  );

  const availableLanguageParams = Object.values(AvailableLanguage).join("|");
  const localesParamString = `/:locale(${availableLanguageParams})`;

  return (
    <div className={classes.root}>
      <Router>
        {navigation}
        <Container maxWidth="md" className={classes.container}>
          <Paper variant="elevation" className={classes.surface}>
            <Switch>
              <Redirect
                exact
                from="/settings"
                to={`/${i18n.language}/settings`}
              />
              <Redirect exact from="/" to={`/${i18n.language}`} />
              {user && (
                <Route path={`${localesParamString}/settings`}>
                  <FirebaseUserDocumentContext.Provider value={docPath}>
                    <EditSettings />
                  </FirebaseUserDocumentContext.Provider>
                </Route>
              )}
              <Route path={`${localesParamString}/`}>{Content}</Route>
            </Switch>
          </Paper>
        </Container>
      </Router>
    </div>
  );
}
