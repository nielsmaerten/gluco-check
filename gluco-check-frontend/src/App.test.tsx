import React from "react";
import { render, cleanup } from "@testing-library/react";
import { axe } from "jest-axe";
import userEvent from "@testing-library/user-event";
import { auth } from "./lib/firebase";
import * as firebaseAuthHooks from "react-firebase-hooks/auth";
import App from "./App";
import { mockUser } from "./lib/__mocks__/firebase";
import { act } from "react-dom/test-utils";

jest.mock("./lib/firebase.ts", () => {
  return {
    auth: {
      signOut: jest.fn(),
    },
  };
});

jest.mock("./pages/EditSettings.tsx", () => {
  return {
    __esModule: true,
    default: () => {
      return <div>EditSettings</div>;
    },
  };
});

jest.mock("./pages/Landing.tsx", () => {
  return {
    __esModule: true,
    default: () => {
      return <div>Landing</div>;
    },
  };
});

jest.mock("./pages/Welcome.tsx", () => {
  return {
    __esModule: true,
    default: () => {
      return <div>Welcome</div>;
    },
  };
});

afterEach(() => {
  jest.clearAllMocks();
  cleanup();
});

describe("App component", () => {
  it("renders the component when user loading", () => {
    // @ts-ignore
    firebaseAuthHooks.useAuthState = jest
      .fn()
      .mockReturnValue([undefined, true]);
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the component when user loaded", async () => {
    // @ts-ignore
    firebaseAuthHooks.useAuthState = jest
      .fn()
      .mockReturnValue([mockUser, false]);
    const { container, findByTestId } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the component when no user and no loading", () => {
    // @ts-ignore
    firebaseAuthHooks.useAuthState = jest
      .fn()
      .mockReturnValue([undefined, false]);
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    // @ts-ignore
    firebaseAuthHooks.useAuthState = jest
      .fn()
      .mockReturnValue([mockUser, false]);
    const { container } = render(<App />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
