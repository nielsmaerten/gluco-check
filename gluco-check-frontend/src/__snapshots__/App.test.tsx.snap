// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the component when no user and no loading 1`] = `
<div
  class="makeStyles-root-17"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-23 MuiToolbar-gutters"
    >
      <section
        class="makeStyles-leftToolbar-20"
      >
        <h1
          class="MuiTypography-root makeStyles-navTitle-24 MuiTypography-h6"
        >
          <a
            href="/en"
          >
            title
          </a>
        </h1>
      </section>
      <section
        class="makeStyles-rightToolbar-21"
      >
        <ul
          class="makeStyles-nav-22"
        >
          <li>
            <div>
              LanguageSelector
            </div>
          </li>
          <li>
            <a
              aria-disabled="false"
              aria-label="navigation.faqs"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-home"
              href="/faq/i18nCode"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              aria-label="navigation.github"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-contribute"
              href="https://github.com/nielsmaerten/gluco-check"
              tabindex="0"
              target="_contribute"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
        </ul>
      </section>
    </div>
  </header>
  <div
    class="MuiContainer-root makeStyles-container-18 MuiContainer-maxWidthMd"
  >
    <div
      class="MuiPaper-root makeStyles-surface-19 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div>
        Landing
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders the component when user loaded 1`] = `
<div
  class="makeStyles-root-9"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-15 MuiToolbar-gutters"
    >
      <section
        class="makeStyles-leftToolbar-12"
      >
        <h1
          class="MuiTypography-root makeStyles-navTitle-16 MuiTypography-h6"
        >
          <a
            href="/en"
          >
            title
          </a>
        </h1>
      </section>
      <section
        class="makeStyles-rightToolbar-13"
      >
        <ul
          class="makeStyles-nav-14"
        >
          <li>
            <div>
              LanguageSelector
            </div>
          </li>
          <li>
            <a
              aria-disabled="false"
              aria-label="navigation.faqs"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-home"
              href="/faq/i18nCode"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              aria-label="navigation.github"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-contribute"
              href="https://github.com/nielsmaerten/gluco-check"
              tabindex="0"
              target="_contribute"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
        </ul>
      </section>
    </div>
  </header>
  <div
    class="MuiContainer-root makeStyles-container-10 MuiContainer-maxWidthMd"
  >
    <div
      class="MuiPaper-root makeStyles-surface-11 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div>
        Welcome
      </div>
    </div>
  </div>
</div>
`;

exports[`App component renders the component when user loading 1`] = `
<div
  class="makeStyles-root-1"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-7 MuiToolbar-gutters"
    >
      <section
        class="makeStyles-leftToolbar-4"
      >
        <h1
          class="MuiTypography-root makeStyles-navTitle-8 MuiTypography-h6"
        >
          <a
            href="/en"
          >
            title
          </a>
        </h1>
      </section>
      <section
        class="makeStyles-rightToolbar-5"
      >
        <ul
          class="makeStyles-nav-6"
        >
          <li>
            <div>
              LanguageSelector
            </div>
          </li>
          <li>
            <a
              aria-disabled="false"
              aria-label="navigation.faqs"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-home"
              href="/faq/i18nCode"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
          <li>
            <a
              aria-disabled="false"
              aria-label="navigation.github"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              data-testid="navigation-contribute"
              href="https://github.com/nielsmaerten/gluco-check"
              tabindex="0"
              target="_contribute"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </li>
        </ul>
      </section>
    </div>
  </header>
  <div
    class="MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthMd"
  >
    <div
      class="MuiPaper-root makeStyles-surface-3 MuiPaper-elevation1 MuiPaper-rounded"
    />
  </div>
</div>
`;
